### ISSUE TITLE
constraint of type parameter should make effect in method declaration

### ISSUE THREAD

**Issue by bvanjoi** [microsoft/TypeScript#61059]
### 🔎 Search Terms

"constraint", "method declaration"

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about "A method and a function property of the same type behave differently"

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.8.0-dev.20250127#code/JYOwLgpgTgZghgYwgAgIIB4AqA+ZBvAKGWRgHtT0BVZCAD0hABMBnZHACloC5lKBKHnBABPAgF8CBGAFcQCMMFIgSABnZweGZmCigA5tgA0yAEab0IaQFsT0bH3xFip5AF5kcZAHovyZgAtSaQAbRmQoCAAHUigwGigoGNNpOMQwaThg4OFkEFIwADpJZ1MCslJ2AEYHH2RZZiCmcUkgA

### 💻 Code

```ts
interface A<T> {
  foo<U extends T>(x: U): any
}

function f0(a: A<string>, b: A<number>) {
    b = a // should report error but actually not.

    b.foo(1) // unsound
}
```

### 🙁 Actual behavior

check success

### 🙂 Expected behavior

`b = a` report `number` cannot assign to `string`.

**Comment by RyanCavanaugh** [comment]
Duplicate #55671 (see #55864)
